src/CMakeLists.txt

```diff
@@ -80,24 +97,27 @@ function(gz_msgs_protoc)
     set(${gz_msgs_protoc_OUTPUT_RUBY_VAR} ${${gz_msgs_protoc_OUTPUT_RUBY_VAR}} PARENT_SCOPE)
   endif()
 
-  #get_cmake_property(_variableNames VARIABLES)
-  #foreach (_variableName ${_variableNames})
-  #    message(STATUS "${_variableName}=${${_variableName}}")
-  #endforeach()
-
   add_custom_command(
-    OUTPUT
-      ${output_files}
-    COMMAND
-      ${gz_msgs_protoc_PROTOC_EXEC}
-    ARGS
-      ${protoc_args}
-      ${ABS_FIL}
+    OUTPUT ${output_files}
+    COMMAND Python3::Interpreter
+    ARGS tools/gz_msgs_generate
+      --protoc-exec "$<TARGET_FILE:${gz_msgs_protoc_PROTOC_EXEC}>"
+      --gz-generator-bin "$<TARGET_FILE:gz_msgs_gen>"
+      --generate-cpp "${gz_msgs_protoc_GENERATE_CPP}"
+      --generate-ruby "${gz_msgs_protoc_GENERATE_RUBY}"
+      --generate-ignition "TRUE"
+      --output-cpp-path "${gz_msgs_protoc_OUTPUT_CPP_DIR}"
+      --output-ruby-path "${gz_msgs_protoc_OUTPUT_RUBY_DIR}"
+      --proto-path "${gz_msgs_protoc_PROTO_PATH}"
+      --input-path "${ABS_FIL}"
     DEPENDS
       ${ABS_FIL}
       gz_msgs_gen
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
```

Comment:
Yellow flag that this operates in the source directory rather than build. Does this modify source contents in place for committing later or is it just easier to work with the source dir paths?

Reply:
Added a note that the working dir i not where the output happens in the cmake